raiden/utils/upgrades.py

```diff
@@ -109,93 +126,78 @@ class UpgradeManager:
+        self._current_version = get_file_version(db_filename)
 
     def run(self):
-        """
-        The `_current_db_filename` is going to hold the filename of the database
-        with the new version. However, the previous version's data
-        is going to exist in a file whose name contains the old version.
-        Therefore, running the migration means that we have to copy
-        all data to the current version's database and execute the migration
-        functions.
-        """
+        # First clear up any partially upgraded databases.
+        #
+        # A database will be partially upgraded if the process receives a
+        # SIGKILL/SIGINT while executing migrations. NOTE: It's very probable
+        # the content of the database remains consistent, because the upgrades
+        # are executed inside a migration, however making a second copy of the
+        # database does no harm.
         paths = glob(f'{self._current_db_filename.parent}/v*_log.db')
-        older_file = older_db_file(paths)
-
-        if older_file is None or older_file == str(self._current_db_filename):
-            return
-
-        old_db_filename = Path(older_file)
-
-        with get_file_lock(old_db_filename), get_file_lock(self._current_db_filename):
-            if self._current_db_filename.exists():
-                db_version = get_db_version(self._current_db_filename)
-
-                # The current version matches our target version, nothing to do.
-                if db_version == RAIDEN_DB_VERSION:
-                    return
+        valid_db_names = filter_db_names(paths)
+        delete_dbs_with_failed_migrations(valid_db_names)
 
-                if db_version > RAIDEN_DB_VERSION:
-                    raise RuntimeError(
-                        f'Database version higher then expected. '
-                        f'It is {db_version} should be {self._current_version}',
-                    )
+        # At this point we know every file version and db version match
+        # (assuming there are no concurrent runs).
+        paths = glob(f'{self._current_db_filename.parent}/v*_log.db')
+        valid_db_names = filter_db_names(paths)
+        latest_db_path = latest_db_file(valid_db_names)
+        file_version = get_file_version(latest_db_path)
 
-                # The version number in the database is smaller then the
-                # current target, this means the migration failed to execute on
-                # the last iteration, delete the partially upgraded database
-                # and start again.
-                self._delete_current_db()
+        # The latest version matches our target version, nothing to do.
+        if file_version == RAIDEN_DB_VERSION:
+            return
 
-            older_version = get_db_version(old_db_filename)
-            if not older_version:
-                # There are no older versions to upgrade from.
-                return
+        if file_version > RAIDEN_DB_VERSION:
```

Comment:
Just a note, another solution to the bug I introduced in the other PR, would be to instead of returning the `highest` version among the `.db`s, one can return the `previous` version, as the one `db` with a version that is lower then the current `RAIDEN_DB_VERSION` (this is the old behavior).

I have not done this because this check would not be possible, and one would be allowed to do a downgrade of the client.